quadruped_pympc/controllers/Controllers.py

```diff
@@ -0,0 +1,543 @@
+        ##############################################
+
+        if(self.controller_type == 'nominal'):
+            from quadruped_pympc.controllers.gradient.nominal.centroidal_nmpc_nominal import Acados_NMPC_Nominal
+            self.controller = Acados_NMPC_Nominal()
+            if self.optimize_step_frequency==True:
+                from quadruped_pympc.controllers.gradient.nominal.centroidal_nmpc_gait_adaptive import Acados_NMPC_GaitAdaptive
+                self.batched_controller = Acados_NMPC_GaitAdaptive()
+
+        elif(self.controller_type == 'input_rates'):
+            from quadruped_pympc.controllers.gradient.input_rates.centroidal_nmpc_input_rates import Acados_NMPC_InputRates
+
+            self.controller = Acados_NMPC_InputRates()
+
+            if self.optimize_step_frequency==True:
+                from quadruped_pympc.controllers.gradient.nominal.centroidal_nmpc_gait_adaptive import Acados_NMPC_GaitAdaptive
+                self.batched_controller = Acados_NMPC_GaitAdaptive()
+
+        elif(self.controller_type ==  'sampling'):
+
+            if self.optimize_step_frequency==True:
+                from quadruped_pympc.controllers.sampling.centroidal_nmpc_jax_gait_adaptive import Sampling_MPC
+
+            else:
+                from quadruped_pympc.controllers.sampling.centroidal_nmpc_jax import Sampling_MPC
+
+            import jax
+            import jax.numpy as jnp
+            num_parallel_computations = self.num_parallel_computations
+            # self.iteration = self.iteration
+            self.controller = Sampling_MPC(horizon = self.horizon, 
+                                      dt = self.mpc_dt, 
+                                      num_parallel_computations = num_parallel_computations,
+                                      sampling_method = self.sampling_method,
+                                      control_parametrization = self.control_parametrization,
+                                      device="gpu")
+            self.best_control_parameters = jnp.zeros((self.controller.num_control_parameters, ))
+            self.jitted_compute_control = jax.jit(self.controller.compute_control, device=self.controller.device)
+            # jitted_get_key = jax.jit(controller.get_key, device=controller.device)
+            self.jitted_prepare_state_and_reference = self.controller.prepare_state_and_reference
+
+            self.index_shift = 0
+
+        # Tracking mpc error (z, x_dot, y_dot, z_dot, roll, pitch, roll_dot, pitch_dot, yaw_dot)
+        self.mean_tracking = np.zeros((9,))
+        self.mean_num_sample = 1
+        self.mean_optimizer_cost = 0
+        self.optimizer_cost = 0
+
+
+
+
+        
+    def periodic_gait_generator(self):
+
+        # Periodic gait generator _________________________________________________________________________
+        gait_name = self.gait_name
+        gait_type, duty_factor, step_frequency = self.get_gait_params(gait_name)
```

Comment:
In the new version, the gait parameters are defined not in a function but rather in the config file. 

Meaning that the function `get_gait_params` should disappear, and rather we should expect to get in the constructor of the class the `gait_params` or rather already the instance of the `PeriodicGaitGenerator`